### ISSUE TITLE
The fortmat processing of 6-dimensional input is incomplete, which will lead to MKLDNNMemoryFormat::undef error.

### ISSUE THREAD

**Issue by baoachun** [PaddlePaddle/Paddle#37990]
https://github.com/PaddlePaddle/Paddle/blob/1911b6f0da63dee423bc47754362da8fc45e1127/paddle/fluid/platform/mkldnn_helper.h#L307

<img width="306" alt="图片" src="https://user-images.githubusercontent.com/22114318/145324775-5447ae83-1eb0-49ac-8f3f-e51985dcc13e.png">

![图片](https://user-images.githubusercontent.com/22114318/145324814-b8487708-506f-4e60-8488-b3a66eb23a37.png)

**Comment by baoachun** [comment]
@lidanqing-intel @jakpiase @jczaja

**Comment by baoachun** [comment]
Export inference model:
```
git clone https://github.com/PaddlePaddle/PaddleClas.git

git checkout develop

bash test_tipc/prepare.sh test_tipc/config/RedNet/RedNet26_train_infer_python.txt whole_infer

python3 tools/export_model.py -c  ppcls/configs/ImageNet/RedNet/RedNet26.yaml
```

Running command:
```
python3 deploy/python/predict_cls.py -c deploy/configs/inference_cls.yaml -o Global.use_gpu=False -o Global.enable_mkldnn=True -o Global.cpu_num_threads=1 -o Global.inference_model_dir=inference/ -o Global.batch_size=16 -o Global.infer_imgs=dataset/ILSVRC2012/val -o Global.benchmark=True
```

**Comment by jakpiase** [comment]
Hi @baoachun.
I have followed your steps and everything works on my setup. Could you please give me more information about system and CPU that you are using?

**Comment by baoachun** [comment]
Hi @jakpiase, the basic information is as follows:
system version: ubuntu 18.04
python version: python3.7
paddle version: the newest develop branch
cpu: 6271

**Comment by jakpiase** [comment]
Hi @baoachun, I have fixed this bug in a PR, but I have an additional question. Is this model pretrained? Even if I run it with `Global.enable_mkldnn=False` I am not getting any meaningful output from it. Is there any simple way to check the correctness of the model's output?

**Comment by baoachun** [comment]
Hi @jakpiase, what do you mean by *meaningful output* ?

**Comment by jakpiase** [comment]
Hi @baoachun, I get the same weird results no matter if I run with mkldnn or without. For example if I run with `python python/predict_cls.py -c configs/inference_cls.yaml -o Global.use_gpu=False -o Global.enable_mkldnn=False  -o Global.cpu_num_threads=1 -o Global.inference_model_dir=../inference/ -o Global.batch_size=4 -o Global.infer_imgs=../dataset/ILSVRC2012/val -o Global.benchmark=True`
 I get only zeros in prediction scores. Is this the desired behavior of the model? Here are the outputs of this command:
```
ILSVRC2012_val_00000001.JPEG:   class id(s): [999, 328, 341, 340, 339], score(s): [0.00, 0.00, 0.00, 0.00, 0.00], label_name(s): ['toilet tissue, toilet paper, bathroom tissue', 'sea urchin', 'hog, pig, grunter, squealer, Sus scrofa', 'zebra', 'sorrel']
ILSVRC2012_val_00000002.JPEG:   class id(s): [999, 328, 341, 340, 339], score(s): [0.00, 0.00, 0.00, 0.00, 0.00], label_name(s): ['toilet tissue, toilet paper, bathroom tissue', 'sea urchin', 'hog, pig, grunter, squealer, Sus scrofa', 'zebra', 'sorrel']
ILSVRC2012_val_00000003.JPEG:   class id(s): [999, 328, 341, 340, 339], score(s): [0.00, 0.00, 0.00, 0.00, 0.00], label_name(s): ['toilet tissue, toilet paper, bathroom tissue', 'sea urchin', 'hog, pig, grunter, squealer, Sus scrofa', 'zebra', 'sorrel']
ILSVRC2012_val_00000004.JPEG:   class id(s): [999, 328, 341, 340, 339], score(s): [0.00, 0.00, 0.00, 0.00, 0.00], label_name(s): ['toilet tissue, toilet paper, bathroom tissue', 'sea urchin', 'hog, pig, grunter, squealer, Sus scrofa', 'zebra', 'sorrel']
```

**Comment by baoachun** [comment]
Hi @jakpiase, I just learned that the model needs to be replaced in order to verify accuracy，you can download the model from https://paddle-imagenet-models-name.bj.bcebos.com/dygraph/inference/RedNet26_infer.tar

**Comment by jakpiase** [comment]
Hi @baoachun, I have checked that with the changes in my PR accuracy stays the same and the error is resolved, so I will do my best to make it merged as soon as possible

**Comment by lidanqing-vv** [comment]
Sorted out locally. Are making PR now.

**Comment by jakpiase** [comment]
Hi @baoachun, every CI except static-check have passed in my PR #38299. This failed CI says that there is code format error, but it doesn't show me in which line does it occur. Do you know how can I get that information? I have used clang-format 3.8.0, and it doesn't see any formatting error

**Comment by jakpiase** [comment]
Nevermind, it has been fixed

**Comment by jakpiase** [comment]
Hi @baoachun, fix PR has been merged already, so could you please confirm that this problem is resolved?

**Comment by baoachun** [comment]
Hi @jakpiase, the problem has been solved, I would like to ask why the data arrangement format can be obtained by the size of the step size？Is there any relevant information to find the basis for this processing method? Thanks!

**Comment by jakpiase** [comment]
Hi @baoachun, what do you mean by "the size of the step size"? Could you please point me to some place in code which interests you?

**Comment by baoachun** [comment]
Hi @jakpiase, just from this https://github.com/PaddlePaddle/Paddle/blob/d1dc677a584393ca9494c2a73596268ed281fc24/paddle/fluid/platform/mkldnn_helper.h#L219

I'm wondering the relationship between the stride and the format tag, and where I can get these information, thanks.

**Comment by jakpiase** [comment]
Hi @baoachun, oneDNN has a nice tutorial for understanding memory format tags and strides: https://oneapi-src.github.io/oneDNN/dev_guide_understanding_memory_formats.html. If you will need some more information, please let me now

**Comment by jakpiase** [comment]
A really short explanation that is skipping most things that are included in oneDNN's explanation:
Let us understand NCHW format as ABCD for simplicity. When we have a tensor with dims: (2, 3, 4, 5) and layout ABCD, the strides are:
stride\[3\](D) = 1
stride\[2\](C) = stride\[3\](D) * dims\[3\] = 5 
stride\[1\](B) = stride\[2\](C) * dims\[2\] = 20
stride\[0\](A) = stride\[1\](B) * dims\[1\] = 60
The same tensor, but with format tag NHWC(ACDB) would have following strides:
stride\[3\](D) = stride\[1\](B) * dims\[1\] = 3
stride\[2\](C) = stride\[3\](D) * dims\[3\] = 15
stride\[1\](B) = 1
stride\[0\](A) = stride\[2\](C) * dims\[2\] = 60

So basically the strides values are the lowest on the rightmost dimension -> D in case of ABCD and B in case of ACDB, and strides values are the highest on the leftmost dimension -> A in both cases. In case of oneDNN's tensor logical dimensions, they are always NCHW(ABCD), no matter what format tag is used, so dims = (2,3,4,5) are always treated as A=2, B=3, C=4, D=5
